tests/transactions/builder/test_vote.py

```diff
@@ -1,121 +1,114 @@
-from solar_crypto.transactions.builder.vote import Vote
+from solar_crypto.transactions.builder.vote import Vote, sort_votes
 
 set_network(Testnet)
 
 
-@pytest.mark.parametrize("version", [2, 3])
-def test_vote_transaction_with_pubkey(version):
-    """Test if a vote transaction gets built"""
-    vote = "+034151a3ec46b5670a682b0a63394f863587d1bc97483b1b6c70eb58e7f0aed192"
+def test_vote_transaction_input_error():
+    vote = {"fun": 50.67, "deadlock": 50.3333}
 
     transaction = Vote()
-    transaction.set_votes([vote])
-    transaction.set_type_group(TRANSACTION_TYPE_GROUP.CORE)
-    transaction.set_nonce(1)
-    transaction.set_version(version)
-    transaction.sign("testing")
-    transaction_dict = transaction.to_dict()
-
-    assert transaction_dict["nonce"] == 1
-    assert transaction_dict["signature"]
-    assert transaction_dict["asset"]["votes"]
-    assert transaction_dict["type"] is TRANSACTION_VOTE
-    assert transaction_dict["typeGroup"] == TRANSACTION_TYPE_GROUP.CORE.value
-    assert transaction_dict["fee"] == 5000000
-
-    transaction.verify()  # if no exception is raised, it means the transaction is valid
+    with pytest.raises(SolarInvalidTransaction) as e:
+        transaction.set_votes(vote)
+    assert str(e.value) == "Only two decimal places are allowed."
 
 
-@pytest.mark.parametrize("version", [2, 3])
-def test_vote_transaction_with_username(version):
-    """Test if a vote transaction gets built"""
-
+@pytest.mark.parametrize("vote", [
+    {"fun": 35.3, "deadlock": 64.7},
+    {"fun": 50, "deadlock": 50},
+    {"fun": Decimal(50), "deadlock": Decimal(50)},
+])
+def test_vote_transaction_multiple_votes(vote):
     transaction = Vote()
-    transaction.set_votes(["+asterix"])
-    transaction.set_type_group(TRANSACTION_TYPE_GROUP.CORE)
+    transaction.set_votes(vote)
     transaction.set_nonce(1)
-    transaction.set_version(version)
     transaction.sign("testing")
     transaction_dict = transaction.to_dict()
 
     assert transaction_dict["nonce"] == 1
     assert transaction_dict["signature"]
-    assert transaction_dict["asset"]["votes"]
-    assert transaction_dict["type"] is TRANSACTION_VOTE
-    assert transaction_dict["typeGroup"] == TRANSACTION_TYPE_GROUP.CORE.value
-    assert transaction_dict["fee"] == 5000000
+    assert transaction_dict["asset"]["votes"] == vote
+    assert transaction_dict["type"] is SOLAR_TRANSACTION_VOTE
+    assert transaction_dict["typeGroup"] == TRANSACTION_TYPE_GROUP.SOLAR.value
+    assert transaction_dict["fee"] == 9000000
 
     transaction.verify()  # if no exception is raised, it means the transaction is valid
 
 
-@pytest.mark.parametrize("version", [2, 3])
-def test_vote_switch_transaction_with_username(version):
-    """Test if a vote transaction gets built"""
+@pytest.mark.parametrize("vote", [
+    {"fun": 50, "deadlock": 55},
+    {"fun": 50},
+])
+def test_vote_transaction_must_equal_100(vote):
+    vote = {"fun": 50, "deadlock": 55}
```

Comment:
Shouldn't this be:
```python
    transaction = Vote()
    transaction.set_votes(vote)
```

Reply:
yup, i'm a moron - commit pushed (i've squashed the last 2 tests commit along with this one given it's all the same piece of work)